File: SDKs/googleapis.com/authorizedbuyersmarketplace/v1/java/lib/src/main/java/openapisdk/models/shared/AdSize.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * AdSize
 * Represents size of a single ad slot, or a creative.
**/
public class AdSize {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("height")
    public String height;
    public AdSize withHeight(String height) {
        this.height = height;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("type")
    public AdSizeTypeEnum type;
    public AdSize withType(AdSizeTypeEnum type) {
        this.type = type;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("width")
    public String width;
    public AdSize withWidth(String width) {
        this.width = width;
        return this;
    }
}

File: SDKs/googleapis.com/authorizedbuyersmarketplace/v1/python/src/sdk/models/shared/adsize.py
import dataclasses
from typing import Optional
from enum import Enum
from dataclasses_json import dataclass_json
from sdk import utils

class AdSizeTypeEnum(str, Enum):
    TYPE_UNSPECIFIED = "TYPE_UNSPECIFIED"
    PIXEL = "PIXEL"
    INTERSTITIAL = "INTERSTITIAL"
    NATIVE = "NATIVE"
    FLUID = "FLUID"


@dataclass_json
@dataclasses.dataclass
class AdSize:
    r"""AdSize
    Represents size of a single ad slot, or a creative.
    """
    
    height: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('height') }})
    type: Optional[AdSizeTypeEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('type') }})
    width: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('width') }})
    
